src/components/organize/tasks/TaskList/index.tsx

```diff
@@ -1,25 +1,28 @@
 import { useIntl } from 'react-intl';
+import { useQuery } from 'react-query';
+import { useRouter } from 'next/router';
 import { Card, List } from '@material-ui/core';
 
+import getCampaignTasks from '../../../../fetching/tasks/getCampaignTasks';
 import TaskListItem from './TaskListItem';
-import { ZetkinTask } from '../../../../types/zetkin';
-
 
 interface TaskListProps {
-    tasks: ZetkinTask[];
     hrefBase: string;
 }
 
-const TaskList = ({ tasks, hrefBase }: TaskListProps): JSX.Element => {
+const TaskList = ({ hrefBase }: TaskListProps): JSX.Element => {
     const intl = useIntl();
-
-    // Sort tasks by status
+    const { campId, orgId } = useRouter().query;
+    const { data: tasks } = useQuery(
+        ['campaignTasks', orgId, campId],
+        getCampaignTasks(orgId as string, campId as string),
+    );
```

Comment:
Why you put the query inside the TaskList component? 
To me it makes sense to have the page get the tasks and pass an array of tasks into the component -- so then the component is just responsible to render the list of tasks, and we can process the tasks array depending on what's needed on the page (upcoming tasks, tasks with deadlines only, etc etc)

But maybe I'm not thinking of other things?

Reply:
I think you have a good point! Currently there isn't a case for it to be modified outside the component... but I'll bring it back out of that component.